deps/jerry/jerry-core/ecma/base/ecma-globals.h

```diff
@@ -809,6 +809,7 @@ typedef struct
                                       *    the byte code is function or regular expression.
                                       *    If function, the other flags must be CBC_CODE_FLAGS...
                                       *    If regexp, the other flags must be RE_FLAG... */
+  ecma_value_t name;                /**< ecma value of function name */
```

Comment:
How about storing the name as string(`char*`) in clang?

Reply:
We also should declare how do we use this property, if we use it directly in JS land, ecma_value_t are proper than `char*`?

Reply:
Add internal property is also ok., if memory is acceptable.
ECMA spec does not say function should have name property, but implementer do. In V8,
`function foo(){] print(foo.name)`
will output foo.

Reply:
That's good feature, let's add it if memory is acceptable :)

Reply:
@algebrait could you add a test case for `foo.name`?